test/node.test.ts

```diff
@@ -1,21 +1,81 @@
 import { describe, beforeAll, afterAll } from "vitest";
+import { fetch, Agent } from "undici";
 import { addTests } from "./_tests.ts";
-import { serve } from "../src/adapters/node.ts";
+import { serve, FastResponse } from "../src/adapters/node.ts";
+import { getTLSCert } from "./_utils.ts";
+import { fixture } from "./_fixture.ts";
 
-describe("node", () => {
-  let server: ReturnType<typeof serve> | undefined;
+// TODO: fix CA!
+process.env.NODE_TLS_REJECT_UNAUTHORIZED = "0";
 
-  beforeAll(async () => {
-    process.env.PORT = "0";
-    server = await import("./_fixture.ts").then((m) => m.server);
-    await server!.ready();
-  });
+const tls = await getTLSCert();
 
-  afterAll(async () => {
-    await server?.close();
-  });
+const testConfigs = [
+  {
+    name: "http1",
+    Response: globalThis.Response,
+  },
+  {
+    name: "http1, FastResponse",
+    Response: FastResponse,
+  },
+  {
+    name: "http2 - allowHTTP1",
+    Response: globalThis.Response,
+    useHttp2Agent: true,
+    serveOptions: { tls, node: { http2: true, allowHTTP1: true } },
+  },
+  {
+    name: "http2, FastResponse - allowHTTP1",
+    Response: FastResponse,
+    useHttp2Agent: true,
+    serveOptions: { tls, node: { http2: true, allowHTTP1: true } },
```

Comment:
Here we badically fallback to http1 handler already coveted what is point of it?

Reply:
That was a test - forgot to remove :)